### PR TITLE
Shuffle by record rather than conformer 

### PR DESCRIPTION
**Pull request by ArnNag**
## Description
Address #39 

## Status
- [x] Ready to go

### CODE REVIEW

File: modelforge/dataset/utils.py [hunk 1]
```diff
@@ -126,6 +126,161 @@ def split(self, dataset: "TorchDataset") -> Tuple[Subset, Subset, Subset]:
...
+
+        # add 1 to all the lengths in round-robin fashion until the remainder is 0
+        for i in range(remainder):
+            idx_to_add_at = i % len(subset_lengths)
+            subset_lengths[idx_to_add_at] += 1
+
+        lengths = subset_lengths
+
+        for i, length in enumerate(lengths):
+            if length == 0:
```
Comment (wiederm) [inline modelforge/dataset/utils.py:261]: this results in an empty dataset, can this be intended behavior?

File: modelforge/tests/test_dataset.py [hunk 2]
```diff
@@ -174,22 +231,22 @@ def test_dataset_generation(dataset):
...
+    train_dataset, val_dataset, test_dataset = RandomRecordSplittingStrategy().split(dataset)
 
     energy = train_dataset[0]["E_label"].item()
     assert np.isclose(energy, -412509.9375)
     print(energy)
 
     try:
-        RandomSplittingStrategy(split=[0.2, 0.1, 0.1])
+        RandomRecordSplittingStrategy(split=[0.2, 0.1, 0.1])
     except AssertionError:
```
Comment (wiederm) [inline modelforge/tests/test_dataset.py:245]: that should be `except AssertionError as e`

**Review by wiederm — APPROVED** [review]
Thank you! This looks really good!

File: modelforge/tests/test_dataset.py (hunk 2 above)
Comment (ArnNag) [reply modelforge/tests/test_dataset.py:245]: sorry, used the online editor since i was working on another commit locally and didn't catch this :(

File: modelforge/dataset/utils.py (hunk 1 above)
Comment (ArnNag) [reply modelforge/dataset/utils.py:261]: Perhaps the user intends to create an empty dataset, but this will warn the user that they are doing so. I copied this from the implementation of `torch.utils.data.random_split`

**Review by wiederm — APPROVED** [review]

File: modelforge/dataset/utils.py (hunk 1 above)
Comment (wiederm) [reply modelforge/dataset/utils.py:262]: Thanks for clarifying
